test/e2e/network_test.go

```diff
@@ -696,10 +696,13 @@ func (tc *testContext) createJob(name, image string, command []string, selector
 	return job, err
 }
 
-// deleteJob deletes the job with the given name
+// deleteJob deletes the job with the given name and the dependent pods
```

Comment:
Why is this required to fix this bug? Please state this in the commit message

Reply:
Before, only the job was deleted as part of the cleanup process, leaving the dependent pods
running the test cluster. With the introduction of the parallel upgrade checker
job, not deleting the pods may result in unreliability while collecting
the number of failed pods with several iterations. 

Thinking this twice is not a hard requirement, and the e2e test can live without it.  Will move it to a separate PR, given it is a good practice to remove all the resources created in the e2e test suite.